### ISSUE TITLE
[TF2] [exploit] VAC can be bypassed with console commands.

### ISSUE THREAD

**Issue by xxmyn** [ValveSoftware/Source-1-Games#3911]
A VAC banned account is able to join a VAC secured server by setting their fps_max to 1, host_timescale to ~25, and sv_cheats to 1, almost 100% of the time.

[relevant link 1](https://www.youtube.com/watch?v=-HWKhiWRRsg)
[relevant link 2](https://www.youtube.com/watch?v=F0uYxHRzWo0)
[relevant link 3](https://www.youtube.com/watch?v=B3vBCoFtYV0)
[relevant link 4](https://www.youtube.com/watch?v=H2Cb0qHZO5E)

**Comment by xnst0** [comment]
HAHAHAHAHAAHAH

**Comment by notbemji** [comment]
BAHAHAHHAHAH

**Comment by AnAkkk** [comment]
It seems to be a race condition with how the Steam validation is performed. ISteamGameServer::BeginAuthSession is called before the player net channel is created and signon state properly set.
It's probably that `ValidateAuthTicketResponse_t` callback must arrive too early, and then CGameClient::Disconnect fails to work.

**Comment by FuckingStealingNames** [comment]
This is also working outside of casual, but I do not wish to tell how it works here, is there someone I can PM to avoid community servers also having this sick move?

**Comment by xxmyn** [comment]
I would assume it works with the exact same method as from my knowledge VAC doesn't work differently on community servers,,,,,,,,,,,,,,

**Comment by FuckingStealingNames** [comment]
> I would assume it works with the exact same method as from my knowledge VAC doesn't work differently on community servers,,,,,,,,,,,,,,

Yes, but also kinda no. Since you include a couple of other commands to actually have an entrance to community servers, since just spamming connect to a server will not go.

A few tests with it and a few hours down the drain, it seems that doing a connection attempt multiple times lets it connect, but I'm not sure that it works if it requests a content download from the server, or if it just bans you for having VAC on the account, since every disconnect message is just the same.

**Comment by FuckingStealingNames** [comment]
But since I thought about this slightly outside the box, I wouldn't assume one would know how to replicate this without some knowledge of the Developer Console and the Source Engine commands.

**Comment by talascio** [comment]
This also connects to the item server???

You'll be locked to 30fps after connecting due to fps_max being restricted while connected.  So using this to temporarily circumvent an old vac to play with friends is not going to be enjoyable.  

That being said, many people probably aren't going to use this for innocent reasons.

**Comment by FuckingStealingNames** [comment]
This also connects to the item server from what I remember in testing, I mean, it always did outside of VAC servers, IIRC. Like the only thing it couldn't do is let you do is trade the items away, or buy anything in the Mann.Co. store.

**Comment by xxmyn** [comment]
> This also connects to the item server???
> 
> You'll be locked to 30fps after connecting due to fps_max being restricted while connected. So using this to temporarily circumvent an old vac to play with friends is not going to be enjoyable.
> 
> fps_max desc isn't true in tf2 either, afaik you cannot change it even in spec

cheats can easily change the value of fps_max after you've joined a game, and it's likely anyone using this exploit would also be cheating.

**Comment by w7rus** [comment]
Clientside Innovations LLC

**Comment by MarcinTPL** [comment]
#Valvecoding

**Comment by Aws0mee** [comment]
Have you ever considered it's because valve developers don't want to be bigots? Preventing banned users from playing would be discriminating against them!

**Comment by xxmyn** [comment]
> Have you ever considered it's because valve developers don't want to be bigots? Preventing banned users from playing would be discriminating against them!

so true

**Comment by gir489returns** [comment]
Now you can say the N word to your VAC bans.

**Comment by sapphonie** [comment]
> This is also working outside of casual, but I do not wish to tell how it works here, is there someone I can PM to avoid community servers also having this sick move?

[here](<https://sappho.io/contact), thanks

**Comment by FuckingStealingNames** [comment]
> > This is also working outside of casual, but I do not wish to tell how it works here, is there someone I can PM to avoid community servers also having this sick move?
> 
> [here](https://sappho.io/contact), thanks

I gotchu

**Comment by AzureWoof** [comment]
Hi. This is still an issue. I ran into someone doing this today. This was someone who was actually already manually game banned, but was able to join the server using this method.

Even if this method didn't work, the matchmaking system still reserves a slot for the banned player, and the player can attempt to reconnect infinitely. Theoretically someone could just make a bunch of banned accounts try to connect to matchmaking servers over and over again to take up player slots that could have otherwise gone to a player that would actually play the game.

Ideally, this should probably be changed so that the matchmaker completely rejects banned accounts from the get-go, instead of just the server handling it. A good first step would be to fix this major oversight before making changes like that, though.

**Comment by gaelcoral** [comment]
> Hi. This is still an issue. I ran into someone doing this today. This was someone who was actually already manually game banned, but was able to join the server using this method.
> 
> Even if this method didn't work, the matchmaking system still reserves a slot for the banned player, and the player can attempt to reconnect infinitely. Theoretically someone could just make a bunch of banned accounts try to connect to matchmaking servers over and over again to take up player slots that could have otherwise gone to a player that would actually play the game.
> 
> Ideally, this should probably be changed so that the matchmaker completely rejects banned accounts from the get-go, instead of just the server handling it. A good first step would be to fix this major oversight before making changes like that, though.

I'm surprised that Valve never opted to use the ban system that matchmaking already has

![Image](https://github.com/user-attachments/assets/c7b0429d-92ec-4417-a81e-0feed5c0b59f)

**Comment by BrokenMartin** [comment]
still not fixed
